Change "CONST" to "_DATA" to fix string inlining

dismantl opened the pull request:
I'm not sure if it's because of the version of VS I'm using (VS Community 2022), but the assembly produced by the compiler puts all the string constants in the `_DATA` segment, rather than a `CONST` segment as referenced in the code. This caused string inlining not to work using the masm_shc utility. When I switched `CONST` to `_DATA`, string inlining worked again.

Thanks for your work on this, the paper was super informative.

dismantl committed:
Changed "CONST" to "_DATA" to fix string inlining

hasherezade commented:
@dismantl - thanks for your report, and I am glad that you like my paper!
it seems in some versions of VS it will use `CONST` , and in other `_DATA` - so just changing CONST to _DATA is not a solution, as it will break compatibility with other versions.  We should rather have:

```cpp
            if (seg_name == "CONST"  || seg_name == "_DATA") {
                in_const = true;
            }
```
in order to cover both cases.

dismantl committed:
Check for both CONST and _DATA for string inlining

This should allow string inlining for different versions of Visual Studio that name the segment differently.

dismantl commented:
Good call. I just made the change so it looks for either _DATA or CONST.

hasherezade commented:
thanks!
